core/services/log/eth_subscriber.go

```diff
@@ -0,0 +1,194 @@
+package log
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/eth"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+type (
+	ethSubscriber struct {
+		ethClient eth.Client
+		config    Config
+		chStop    chan struct{}
+	}
+)
+
+func newEthSubscriber(ethClient eth.Client, config Config, chStop chan struct{}) *ethSubscriber {
+	return &ethSubscriber{
+		ethClient: ethClient,
+		config:    config,
+		chStop:    chStop,
+	}
+}
+
+func (sub *ethSubscriber) backfillLogs(latestHeadInDb *models.Head, addresses []common.Address, topics []common.Hash) (chBackfilledLogs chan types.Log, abort bool) {
+	if len(addresses) == 0 {
+		ch := make(chan types.Log)
+		close(ch)
+		return ch, false
+	}
+
+	ctx, cancel := utils.ContextFromChan(sub.chStop)
+	defer cancel()
+
+	utils.RetryWithBackoff(ctx, func() (retry bool) {
+		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		defer cancel()
+
+		latestBlock, err := sub.ethClient.HeaderByNumber(ctx, nil)
+		if err != nil {
+			logger.Errorw("Log subscriber backfill: could not fetch latest block header", "error", err)
+			return true
+		} else if latestBlock == nil {
+			logger.Warn("got nil block header")
+			return true
+		}
+		currentHeight := uint64(latestBlock.Number)
+
+		// Backfill from `backfillDepth` blocks ago.  It's up to the subscribers to
+		// filter out logs they've already dealt with.
+		fromBlock := currentHeight - sub.config.BlockBackfillDepth()
```

Comment:
Weren't we discussing a move to using the latest head instead? or REPLAY_FROM_BLOCK?

Reply:
it does use latest head - just below you have:
```
if latestHeadInDb != nil && uint64(latestHeadInDb.Number) > fromBlock {
			logger.Infow("LogBroadcaster: Using the latest stored head a limit of backfill", "blockNumber", latestHeadInDb.Number, "blockHash", latestHeadInDb.Hash)
			// if the latest head stored in DB is newer, we use it instead as the limit of backfill
			fromBlock = uint64(latestHeadInDb.Number)
		}
```
as for env variables, `we agreed to keep BLOCK_BACKFILL_DEPTH and remove REPLAY_FROM_HEAD`, quoting Spook

Reply:
Ah, got it.